src/components/mixins/ExplorePageMixin.ts

```diff
@@ -0,0 +1,139 @@
+import Vue from 'vue';
+import { Component, Mixins, Prop, Ref, Watch } from 'vue-property-decorator';
+import { mixins, WALLET_CONSTS } from '@soramitsu/soraneo-wallet-web';
+import { SortDirection } from '@soramitsu/soramitsu-js-ui/lib/components/Table/consts';
+import SScrollbar from '@soramitsu/soramitsu-js-ui/lib/components/Scrollbar';
+
+import { getter } from '@/store/decorators';
+
+import type { RegisteredAccountAssetWithDecimals } from '@/store/assets/types';
+
+@Component
+export default class ExplorePageMixin extends Mixins(
+  mixins.LoadingMixin,
+  mixins.PaginationSearchMixin,
+  mixins.FormattedAmountMixin
+) {
+  readonly FontSizeRate = WALLET_CONSTS.FontSizeRate;
+  readonly FontWeightRate = WALLET_CONSTS.FontWeightRate;
+
+  @Ref('table') readonly tableComponent!: any;
+  @Prop({ default: '', type: String }) readonly exploreQuery!: string;
+  @Watch('exploreQuery')
+  private resetCurrentPage(): void {
+    this.currentPage = 1;
+  }
+
+  @getter.wallet.account.isLoggedIn isLoggedIn!: boolean;
+  @getter.assets.assetDataByAddress public getAsset!: (addr?: string) => Nullable<RegisteredAccountAssetWithDecimals>;
+
+  order = '';
+  property = '';
+
+  get loadingState(): boolean {
+    return this.parentLoading || this.loading;
+  }
+
+  get pricesAvailable(): boolean {
+    return Object.keys(this.fiatPriceAndApyObject).length > 0;
+  }
+
+  get isDefaultSort(): boolean {
+    return !this.order || !this.property;
+  }
+
+  get preparedItems(): any[] {
+    console.warn('[ExplorePageMixin]: "preparedItems" computed property is not implemented');
+    return [];
+  }
+
+  get filteredItems() {
+    const search = this.exploreQuery.toLowerCase().trim();
+
+    if (!search) return this.preparedItems;
+
+    const filterAsset = (asset): boolean =>
+      asset?.name?.toLowerCase?.()?.includes?.(search) ||
+      asset?.symbol?.toLowerCase?.()?.includes?.(search) ||
+      asset?.address?.toLowerCase?.() === search;
+
+    return this.preparedItems.filter(
+      (item: any) =>
+        filterAsset(item) ||
+        filterAsset(item.poolAsset) ||
+        filterAsset(item.rewardAsset) ||
+        filterAsset(item.baseAsset) ||
+        filterAsset(item.targetAsset)
+    );
+  }
+
+  get sortedItems() {
+    if (!this.order || !this.property) return this.filteredItems;
```

Comment:
could we reuse this getter `isDefaultSort`? If not please use `!(this.order && this.property)`